typespec-tests/src/test/java/com/type/property/additionalproperties/SpreadTests.java

```diff
@@ -0,0 +1,286 @@
+        propertyMap.put("prop", new ModelForRecord("ok"));
+        DifferentSpreadModelRecord body = new DifferentSpreadModelRecord("abc");
+        body.setAdditionalProperties(propertyMap);
+        spreadDifferentModelClient.put(body);
+
+        DifferentSpreadModelRecord record = spreadDifferentModelClient.get();
+        Assertions.assertNotNull(record);
+        Assertions.assertEquals("abc", record.getKnownProp());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop"));
+        Assertions.assertEquals("ok", record.getAdditionalProperties().get("prop").getState());
+    }
+
+    @Test
+    @Disabled("The 'Get' request did not respond for a long time.")
+    public void testSpreadDifferentModelArray() {
+        Map<String, List<ModelForRecord>> propertyMap = new LinkedHashMap<>();
+        propertyMap.put("prop", Arrays.asList(new ModelForRecord("ok"), new ModelForRecord("ok")));
+        DifferentSpreadModelArrayRecord body = new DifferentSpreadModelArrayRecord("abc");
+        body.setAdditionalProperties(propertyMap);
+        spreadDifferentModelArrayClient.put(body);
+
+        DifferentSpreadModelArrayRecord record = spreadDifferentModelArrayClient.get();
+        Assertions.assertNotNull(record);
+        Assertions.assertEquals("abc", record.getKnownProp());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop"));
+        record.getAdditionalProperties().get("prop").forEach(modelForRecord ->
+                Assertions.assertEquals("ok", modelForRecord.getState()));
+    }
+
+    @Test
+    @Disabled("The body provided of 'put' request doesn't match expected body")
+    public void testSpreadRecordUnion() {
+        Map<String, BinaryData> propertyMap = new LinkedHashMap<>();
+        propertyMap.put("prop1", BinaryData.fromString("abc"));
+        propertyMap.put("prop2", BinaryData.fromObject(43.125));
+        SpreadRecordForUnion body = new SpreadRecordForUnion(true);
+        body.setAdditionalProperties(propertyMap);
+        spreadRecordUnionClient.put(body);
+
+        SpreadRecordForUnion record = spreadRecordUnionClient.get();
+        Assertions.assertNotNull(record);
+        Assertions.assertTrue(record.isFlag());
+        Assertions.assertNotNull(record.getAdditionalProperties());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop1"));
+        Assertions.assertEquals("\"abc\"", record.getAdditionalProperties().get("prop1").toString());
+        Assertions.assertNotNull(record.getAdditionalProperties().get("prop2"));
+        Assertions.assertEquals(43.125, Double.parseDouble(record.getAdditionalProperties().get("prop2").toString()));
+    }
+
+    @Test
+    @Disabled("The body provided of 'put' request doesn't match expected body")
```

Comment:
This is your test problem.

These should be `BinaryData.fromObject(new WidgetData2())` or `BinaryData.fromObject(new WidgetData1())`

See source https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/type/property/additional-properties/main.tsp#L531-L536

Reply:
Same for the getter part. It need to be converted to `WidgetData1` with `BinaryData.toObject`

Reply:
There is a bug in code that make the "put" part not working, though.

Hopefully there be a relatively easy fix.

Reply:
Fixed in PR https://github.com/Azure/autorest.java/pull/2699

Please refresh main after it merged, and try enable those tests about `BinaryData`.

I've also use one of your test in that PR (with changes). Please check that as well. Basically you should always use `fromObject` and `toObject` on `BinaryData`.

Reply:
Fixed in the new version.

Reply:
Why is this test still `@Disabled`?

Reply:
Enabled the test cases except `*ModelArray`.